.changeset/tricky-berries-begin.md

```diff
@@ -0,0 +1,22 @@
+---
+'@pandacss/generator': minor
+---
+
+Added a new type to extract variants out of styled components
+
+```tsx
+import { VariantStyledProps } from '../styled-system/jsx'
+
+const Button = styled('button', {
+  base: { color: 'black' },
+  variants: {
+    state: {
+      error: { color: 'red' },
+      success: { color: 'green' },
+    },
+  },
+});
+
+type ButtonVariantProps = VariantStyledProps<typeof Button>;
```

Comment:
small nit, I'd personally prefer having a naming a bit more consistent with what we have currently (`RecipeVariantProps`), what do you think of `InferStyledVariantProps` ?

the rest of the PR looks good to me !

Reply:
I'm voting for either `StyledVariantProps<typeof Button>` or both 🙋🗳️

Reply:
So I used `VariantStyledProps` because the HTML type that uses `StyledProps` as the suffix `HTMLStyledProps`, sounded more consistent, I rename to the suggested one 👌